ops/pebble.py

```diff
@@ -1483,6 +1492,51 @@ class NoticesUsers(enum.Enum):
     """
 
 
+class ChangeKind(enum.Enum):
+    """Enum of Pebble Change kinds."""
+
+    START = 'start'
+    STOP = 'stop'
+    RESTART = 'restart'
+    REPLAN = 'replan'
+    EXEC = 'exec'
+    RECOVER_CHECK = 'recover-check'
+    PERFORM_CHECK = 'perform-check'
+
+
+class ChangeStatus(enum.Enum):
+    """Enum of Pebble Change statuses."""
+
+    HOLD = 'hold'
+    """Hold status means the task should not run for the moment, perhaps as a
+	consequence of an error on another task."""
+
+    DO = 'do'
+    """Do status means the change or task is ready to start."""
+
+    DOING = 'doing'
+    """Doing status means the change or task is running or an attempt was made to run it."""
+
+    DONE = 'done'
+    """Done status means the change or task was accomplished successfully."""
+
+    ABORT = 'abort'
+    """Abort status means the task should stop doing its activities and then undo."""
+
+    UNDO = 'undo'
+    """Undo status means the change or task should be undone, probably due to an error."""
+
+    UNDOING = 'undoing'
+    """UndoingStatus means the change or task is being undone or an attempt was made to undo it."""
+
+    ERROR = 'error'
+    """Error status means the change or task has errored out while running or being undone."""
+
+    WAIT = 'wait'
+    """Wait status means the task was accomplished successfully but some
```

Comment:
This is briefly mentioned in the PR description. You shouldn't generally need to use these in the charm code - Juju does the work of figuring out that it's (perform-check, error) or (recover-check, done) and translates that into an appropriate event. We don't even give the notice key/change ID, so querying Pebble for the raw data would be tricky.

You do need to use them if you're writing Harness tests (as opposed to Scenario, where it will be much simpler). For that, you only really need those two combinations.

However, we do expose `get_notice` and `get_check` in `ops.pebble.Client` so it seems more correct to me that we let people use an enum rather than use a bunch of strings. It's quite a few lines of code, but should be nearly no maintenance burden (it seems unlikely the list of statuses will change, and if there are new change kinds then that's probably something we would want to add anyway) and doesn't change performance in any way. It's a few more options in an autocomplete, but I don't think that will really be impacting people.

I did expect push back on this, and I don't feel super strongly about it, so please do outline why it would be better to not do it if you disagree.

Reply:
I thought about suggesting we only add the ones we need, but I think this is fine as is. It's just enum values, not complicated code, and these are very unlikely to change.

Reply:
I don't have a strong opinion against inclusion of these values.
I would maybe consider documenting which of these aren't expected to be visible to the ops API user, so that they don't write huge dispatch functions for every possible value.